src/static/js/cockpit.js

```diff
@@ -11,21 +11,47 @@
 
   //Cockpit is inheriting from EventEmitter2.
   var Cockpit = function Cockpit(csocket) {
-    this.socket = csocket;
+    var self = this;
+    this.uiLoader = new window.UiLoader();
+    this.rov = new window.MessageManager(csocket);
     this.sendUpdateEnabled = true;
     this.capabilities = 0;
     this.loadedPlugins = [];
-    this.loadPlugins();
-    console.log('loaded plugins');
-    // Register the various event handlers
-    this.listen();
+
+    this.loadUi(function() {
+      self.extensionPoints = {
+        rovSettings: $('html /deep/ rov-settings'),
+        rovDiagnostics: $('html /deep/ rov-diagnostics /deep/ #dropIn'),
+        videoContainer: $('html /deep/ rov-video'),
+        keyboardInstructions: $('html /deep/ #keyboardInstructions'),
+        buttonPanel: $('html /deep/ #buttonPanel'),
+        menu: $('html /deep/ rov-menu'),
+        inputController: undefined //will be set by the plugin
```

Comment:
maybe we just move all the UI elements into ```extensionPoints.UI.````
The intention of having the inputController there (and subsequently the headsUpMenu once it's loaded) is to have one place to go to for adding your functionality when you develop a plugin.